VM/rel2vm_params.py

```diff
@@ -0,0 +1,1085 @@
+
+
+def optimise_vm_params(
+    srf_meta, options, out_dir, ptemp, logger: Logger = qclogging.get_basic_logger()
+):
+    """
+    Optimises VM domain and returns a dictionary that extends vm_params_dict
+
+    Parameters
+    ----------
+    srf_meta : Data extracted from REL csv file
+    options : Default/User-specified options (via arguments)
+    out_dir : Directory where output files are eventually saved (eg. ..../Data/VMs/Hossack)
+    ptemp : Temporary work directory (eg. .../Data/VMs/Hossack/_tmp_Hossack_nho8ui41)
+    logger :
+
+    Returns
+    -------
+    vm_params_dict_extended that contains extra info not stored in vm_params.yaml
+
+    """
+    # properties stored in classes (fault of external code)
+    faultprop.Mw = srf_meta["mag"]
+    faultprop.rake = srf_meta["rake"]
+    faultprop.dip = srf_meta["dip"]
+    faultprop.faultstyle = None
+    # rrup to reach wanted PGV
+    if options["pgv"] == -1.0:
+        pgv = mag2pgv(faultprop.Mw)
+        logger.debug("pgv determined from magnitude. pgv set to {}".format(pgv))
+    else:
+        pgv = options["pgv"]
+        logger.debug("pgv set to {}".format(pgv))
+    rrup, pgv_actual = find_rrup(pgv)
+
+    if "dtop" in srf_meta:
+        fault_depth = np.min(srf_meta["dtop"])
+    else:
+        fault_depth = srf_meta["hdepth"]
+
+    rjb = 0
+    if fault_depth < rrup * 2:
+        # rjb = (rrup ** 2 - fault_depth ** 2) ** 0.5
+        rjb = max(
+            options["min_rjb"], rrup
+        )  # sets rrup equal to rjb to ensure deep ruptures have sufficient VM size
+
+    # original, unrotated vm
+    bearing = 0
+    origin, xlen0, ylen0 = determine_vm_extent(
+        rjb, options["hh"], srf_meta["corners"].reshape((-1, 2)), rot=bearing, wd=ptemp
+    )
+
+    if fault_depth > rrup:
+        logger.warning(
+            "fault_depth > rrup. Fault too deep, will not generate VM. "
+            f"Setting xlen, ylen to 0. Previous values: x:{xlen0}, y:{ylen0}"
+        )
+        xlen0 = 0
+        ylen0 = 0
+
+    o1, o2, o3, o4 = build_corners(origin, bearing, xlen0, ylen0)
+    vm0_region = corners2region(o1, o2, o3, o4)
+    plot_region = (
+        vm0_region[0] - 1,
+        vm0_region[1] + 1,
+        vm0_region[2] - 1,
+        vm0_region[3] + 1,
+    )
+
+    # proportion in ocean
+    if options["no_optimise"] or xlen0 <= 0 or ylen0 <= 0:
+        logger.debug("Not optimising for land coverage")
+        land0 = 100
+    else:
+        land0 = vm_land(o1, o2, o3, o4, wd=ptemp)
+        logger.debug("Land coverage found to be {}%".format(land0))
+
+    # for plotting and calculating VM domain distance
+    with open("%s/srf.path" % (ptemp), "wb") as sp:
+        for plane in srf_meta["corners"]:
+            sp.write("> srf plane\n".encode())
+            np.savetxt(sp, plane, fmt="%f")
+            sp.write("%f %f\n".encode() % (tuple(plane[0])))
+
+    # modify VM if necessary
+    adjusted = False
+    if faultprop.Mw >= 3.5 and land0 < 99:
```

Comment:
I think we recently transitioned to doing optimisation all the time. But it could be a good idea as it in-effect gives you an easy way to turn off optimisation.

Reply:
target_land_coverage introduced - to avoid confusion, either no_optimise or target-land-coverage can be set.